bpf/lib/maglev.h

```diff
@@ -0,0 +1,153 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+/* Copyright (C) 2016-2020 Authors of Cilium */
+
+#ifndef __MAGLEV_H_
+#define __MAGLEV_H_
+
```

Comment:
Are you aware of any other features or properties that this implementation provides which are not covered by the current implementation in the `master` branch (from PR #13131)?

Reply:
Awesome work @lrouter! From a quick glance it looks like the core bits of the implementation are very similar with  #13131 which was merged ~5 days ago. You do have some additional agent bits and even more test cases. Would it be possible to rebase, clean up & integrate these against latest master so we can review the missing bits and get them integrated? That would be really nice. We should also remove the vendor'ed murmur3 and switch to your implementation. Looking forward!

Reply:
@borkmann Sure. I will rebase it based on the latest master.